client/state/jitm/selectors.js

```diff
@@ -23,12 +23,33 @@ export const getJITM = ( state, messagePath ) =>
  * @param {string} messagePath The jitm message path (ex: calypso:comments:admin_notices)
  * @returns {object} A jitm
  */
+
+let JITMCache = [];
+
 export const getTopJITM = ( state, messagePath ) => {
 	const jitms = getJITM( state, messagePath );
 
-	if ( jitms.length === 0 ) {
+	// If we have yet to populate the JITM state for the
+	// given message path then return from the previous JITM
+	// cache until JITMS can be fulfilled from network.
+	if ( jitms === null ) {
```

Comment:
If `jitms` is `null` then we have yet to fulfill JITMs from the network. Therefore we can fallback to providing the previous cached JITMs until the request is fulfilled.